client/default.nix

```diff
@@ -1,11 +1,22 @@
-{ mkDerivation, aeson, base, lens, containers, miso, servant, stdenv }:
-mkDerivation {
-  pname = "miso-isomorphic-example";
-  version = "0.1.0.0";
-  src = ../.;
-  isLibrary = false;
-  isExecutable = true;
-  executableHaskellDepends = [ aeson base containers miso lens servant ];
-  description = "Example showing the isomorphic aspect of miso";
-  license = stdenv.lib.licenses.unlicense;
-}
+{ pkgs ? import ../nixpkgs.nix }:
+
+let
+
+  _client = pkgs.haskell.packages.ghcjs.callCabal2nix "client" ./. {
+    common = pkgs.haskell.packages.ghcjs.callCabal2nix "common" ../common {};
+  };
+
+  _client_pkg = pkgs.stdenv.mkDerivation {
+    name = "client";
+    src = ./.;
+    installPhase = ''
+      mkdir -p $out/static
+      ${pkgs.closurecompiler}/bin/closure-compiler ${_client}/bin/client.jsexe/all.js > $out/static/all.js
```

Comment:
Making a separate package to run the closure compiler is actually a nifty idea. It might be a good idea, though, to override the original package. That way, you should end up with only *one* store path for the package.